### DISCUSSION TITLE
WWA shapefiles contain future issuance dates

### DISCUSSION THREAD

**Discussion by greenlaw** [weather-gov/api#387] [Q&A]
**Describe the bug**
The attribute tables of CAP1.2/HCE-derived Watches/Warnings/Advisories (wwa) shapefiles contain issuance dates in the future which do not match the issuance dates shown on the text product.

**To Reproduce**
Download shapefile at https://tgftp.nws.noaa.gov/SL.us008001/DF.sha/DC.cap/DS.WWA/current_all.tar.gz  and inspect attribute table (e.g. using QGIS or ArcGIS).

**Expected behavior**
`issuance` field values should match the text product.

**Screenshots**
![image](https://user-images.githubusercontent.com/1404098/68333057-d9afc380-00a5-11ea-8189-24ff1b340d9d.png)
-> Corresponding text product URL:  https://alerts-v2.weather.gov/products/NWS-IDP-PROD-3903035-3323499

**Environment**
N/A

**Additional context**
I am guessing the `issuance` field is being populated from the VTEC Event Start time rather than the WWA issuance time.  May want to double-check the `expiration` field as well.

**Comment by scadergit** [comment]
Thanks for your patience while I investigated this descreptancy. The shapefile specification was developed by the NWS without regard to the CAP spec. The "ISSUANCE" field in the shapefile is based on the event VTEC and correlates to the CAP "onset" property, not the name-matched CAP "issuance" property. Changing this will require an service change notice (SCN), and likely won't be a high priority. Please simply take note of this discreptancy when parsing the shapefile.

**Comment by greenlaw** [comment]
Thanks for the update.  I understand it might not be the highest priority, but in its current state it's misrepresenting some fairly important information.  We currently rely on the issuance field in our processing and in our map service configuration to add confidence that only active, recently-updated hazard information is being presented to users, so this is likely going to hold up our switch from the CRH shapefile to this product.

Were you also able to confirm how the expiration field is being populated?

As a side note, I think including separate dedicated fields for "onset" or other CAP time values would be a good addition.

**Comment by StephenClouse** [comment]
Unfortunately the data specification of this shapefile is not something we're permitted to change. They were built to a specification from another internal group to support another project.

I can answer the other two questions since I wrote the program that generates the shapefile:

1. It retrieves all of its data from the API /alerts/active endpoint, and thus it should only contain alerts that are actually valid (subject to the update interval for the shapefile itself, which should be 2 minutes for the short-term warnings file and 10 minutes for all others).

1. The expiration time is the VTEC event ending time.

**Comment by greenlaw** [comment]
Thanks Stephen for the info.

Again I understand there may be other pressing matters & the need to go through the SCN process, but it seems that either the original data specification or the implementation was incorrect.  It's my understanding that these two shapefiles were meant to replace the ones being generated by CRH.  In those shapefiles, issuance and expiration represent actual product issuance and expiration, not the VTEC onset/end time, so IMO this really should be classified as a bug fix rather than a product change.  If you need any further justification or information from me, let me know.

**Comment by scadergit** [comment]
I verified the implementation was correct, and the data specification is inconsistent but correct in the context of the purpose that defined it. It was not our decision to use a specification that insufficently replaces the one generated by CRH. The best option would be to follow-up with your source (presumably a weather office or region) to raise this concern through the proper channels. Happy to help make connections as necessary.

**Comment by greenlaw** [comment]
Well I don't have access to the specs involved, so I have no knowledge of how they were written up, but it's pretty clear there is an issue here.  I'm just trying to help identify the problem and get it addressed.  I'll try following up through DISS / GIS.

**Comment by scadergit** [comment]
GIS simply imports the shapefile noted in the original report (compensating for the noted discrepancy). I would reach out to paul.kirkwood@noaa.gov, he's largely involved with the shapefile specification. Feel welcome to cc me using the noaa address.

**Comment by greenlaw** [comment]
Will do, thanks.
